### DISCUSSION TITLE
Authentication failed for git push

### DISCUSSION THREAD

**Discussion by ulugbey21** [community/community#38705] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Dear,

I try to push a text to github reprository. The following commands I typed;

``
git init
git add .
git status
git commit -m "First commit"
git remote add origin https://github.com/ulugbey21/modi_singlephase_viscous.git
git push -u origin master
``
After typing these commands, it prompts to enter my username and password. After entering them, it gives the following error

``
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/ulugbey21/modi_singlephase_viscous.git/'

``
Do you know what can be reason of this authentication error? 

Kind regards,

**Comment by airtower-luna** [comment, marked as answer]
You cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by rivondave** [reply]
Very helpful, thanks

**Reply by slarson2324** [reply]
What happens if the personal access token is not working as a verification method? It's set to public, but when I correctly enter my username and personal access token, I still get 'remote: Invalid username or password. fatal: Authentication failed for '<repo link>'.

**Reply by rivondave** [reply]
It has to work, can you send a screenshot ?

**Reply by Thuledu** [reply]
THANK YOU SOOOOOOO MUCH!!!!

**Reply by AbdulBareeh102** [reply]
Mine is not working even after get the token

**Reply by gustvrcm** [reply]
Using the access token as password was not working for me too as I typed the username and password in the window that appeared. But eventually, after the error message that came in the command prompted, I entered the username/token in the command prompt itself (a login is prompted in the command prompt after the error) and it worked.

**Reply by SimonDedman** [reply]
@gustvrcm please can you explain exactly where you saw that login prompt? In RStudio I get the dialog box and nothing happens after it fails, I just have to close it. In linux terminal, after the 'fatal' message it just takes me back to the command prompt again. Thanks.

**Reply by Kunal-Yelekar** [reply]
Thank you so much @airtower-luna.

**Reply by Shivanigilla** [reply]
Thanks a lot, It worked

**Comment by ulugbey21** [comment]
Thanks. The problem is solved thanks to your suggestion.

**Comment by airtower-luna** [comment]
Glad I could help! Marking the answer would be appreciated. :slightly_smiling_face:

**Comment by sushobhan1993** [comment]
Yes, it works. Thank you

**Comment by dariohsb** [comment]
You need to use your personal token as password

**Comment by elbrant** [comment]
1. [switch from HTTPS to SSH](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#switching-remote-urls-from-https-to-ssh)
2. [change your user.email config to your "github noreply" email address (if you wanted to keep your email private)](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#switching-remote-urls-from-https-to-ssh)
3. navigate into your email settings on github: login > click on your avatar in the upper right corner > settings > email
4. uncheck the "Block command line pushes that expose my email" checkbox

And you should be good to go!!
